### ISSUE TITLE
Let end user specify keyboard shortcut

### ISSUE THREAD

**Issue by Skeletonxf** [Skeletonxf/duplicate-tab#2]
# Customisable shortcuts on Duplicate Tab and Image Extract

Doesn't seem available yet

https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/commands

Appears that the only way to create commands to then register listeners to is to use the manifest.json at the moment.

**Comment by Skeletonxf** [comment]
What I can do is provide several sensible shortcuts for users to pick from. For that I need to know what alternative shortcuts users want so if you don't like CTRL+SHIFT+D please reply here with what would work better for you.

**Comment by stonecrusher** [comment]
I hope that Mozilla will deliver shortcut API at some time because many extension used to have helpful shortcuts which now overlap.

For me personally `Shift + Alt + D` would fit well as I already have a global hotkey on `Shift + Ctrl + D`.

Thanks for the extension anyway!

p.s.: FF57+ already has an item to duplicate tab in tab context menu

**Comment by Skeletonxf** [comment]
Added `SHIFT + ALT + D` to 1.6
I know there's already a duplicate tab context menu now :P That's why it's not enabled by default anymore.

**Comment by stonecrusher** [comment]
Cool! And thanks to the WebExtensions format it's already reviewed and live on amo.
Unfortunately it doesn't work for me.

If I disable the program that occupied `ctrl + shift + D` and press that combination, FF opens something bookmark-related [(see here)](https://www-archive.mozilla.org/docs/end-user/moz_shortcuts.html#nav_bookmark) and `alt + shift + D` doesn't do anything at all.
Of course I enabled the option respectively, even restarted FF afterwards. Also tried fresh profile and in a VM.

So in short: Both shortcuts don't work for me :( `Win 8.1` `FF 57.0.1 (x64)` `Duplicate tab 1.6`
I guess I have to wait until they provide the needed API. You're not the only one struggling with shortcuts atm

**Comment by Skeletonxf** [comment]
:/

Strangely `CTRL + SHIFT + D` and `ALT + SHIFT + D` work for me on Ubuntu. I also tested on a Mac briefly to verify `OPTION + SHIFT + D` worked for them, though other mac users on the reviews reported `CTRL + SHIFT + D` not working for them either.

It's a relief that I'm not the only one.

I suppose for now you can look into AutoHotKey to script automatically clicking the icon.

**Comment by Skeletonxf** [comment]
Update on this feature:
https://www.reddit.com/r/firefox/comments/7x11hu/firefox_60_allows_webextensions_to_customize/

https://github.com/piroor/webextensions-lib-shortcut-customize-ui

https://bugzilla.mozilla.org/show_bug.cgi?id=1421811

**Comment by Skeletonxf** [comment]
Shortcut customization branch now has working customisation for shortcuts: https://github.com/Skeletonxf/duplicate-tab/tree/shortcut-customization

Need to do more testing on Windows to check the defaults at least mostly work, and gracefully handle Firefox versions <60 where customization isn't possible.

**Comment by stonecrusher** [comment]
So long I am staying with release version, so I can tell you in a few months if everything works as intended :P 
But I'm really happy and looking forward to it. Nothing more difficult than changing old habits.

**Comment by Skeletonxf** [comment]
I intend to push the changes onto amo much sooner than in a few months :P

**Comment by stonecrusher** [comment]
Yeah sorry, I should have been more specific. I was talking about the FF60 release which will be on 2018-05-09. Before it won't work :-/

**Comment by Skeletonxf** [comment]
Customization in browser is now on AMO.

The customize library generated a few warnings of `This API has not been implemented by Firefox.` which I suppose makes sense given they only work in Nightly at the moment, and are undocumented on MDN.

I will leave this issue open till the API is actually documented or reaches a more stable version of Firefox in case the API breaks.

**Comment by Skeletonxf** [comment]
Works in Firefox Developer edition now
